### ISSUE TITLE
widget.destroy() fails if widget was not fully initialized before being cleaned up

### ISSUE THREAD

**Issue by GothikX** [kendo-labs/knockout-kendo#128]
The title doesn't really do this justice as it's a very specific edge case... let me try again.

I'm seeing this ONLY with the latest version of kendo. Now I know you say in readme.md that it's not supported but you added bindings for widgets that have only been added in this version soooo... :)

See example here: http://jsfiddle.net/AlexPaven/q5b8C/

Suppose you have a kendo binding that in its template contains another kendo binding; if you try to remove the elements (in my example using an if binding), the inner kendo widget is not properly... initialized, I suppose? Anyway, knockout-kendo attempts to call widget.destroy() and that fails, because widget.element is null. 

Now, of course it could be easily solved by just checking if widget.element is there before calling destroy()... but of course I'd like to know why this happens first. Maybe knockout-kendo is not initializing nested widgets properly? Or is it something that should be fixed in kendo? I'll take another look at this tomorrow and update this if I find more info. For now, to get my application working, I'll probably modify my copy of knockout-kendo to check the element is there before calling destroy.

**Comment by mishoo** [comment]
I could not see any errors with your sample, even when using the latest Kendo UI code.  However, I added a check for `widget.element` before calling destroy, as this can't hurt.

Similar happened in Angular bindings — we setup a element `$destroy` handler which calls `widget.destroy()`, because otherwise there would be a memory leak; however, an outer widget destroys all inner widgets anyway, so the bindings would end up calling destroy for the inner widgets a second time.  For now [calling destroy() twice isn't supported](https://github.com/telerik/kendo-ui-core/issues/99).
